File: selection_sort/selection_sort.c
#include<stdio.h>

int main()
{	
	int N, i, j, k;
	printf("Number of elements in list : ");
	scanf("%d", &N);
	
	int list[N];
	printf("Enter the values in list : ");
	for(i=0; i<N; i++){
		scanf("%d", &list[i]);
	}
    
    int min;
    for(j=0; j<N-1; j++){
    	min=j;
	    for(i=j+1; i<N; i++){
			if(list[min]>list[i]){
    			min=i;
			}   	
		}
		int val;
		val=list[min];
		list[min]=list[j];
		list[j]=val;
	}
	
	for(i=0; i<N; i++){
		printf("%d ", list[i]);
	}	
	
    return 0;
}

File: selection_sort/SelectionSort.java
import java.util.Scanner;
public class SelectionSort 
{
   public static void main(String args[])
   {
		Scanner sc = new Scanner(System.in);
        System.out.print("Enter the length of the array:");
		int size = sc.nextInt();
		int [] array = new int[size];
		System.out.println("Enter the elements of the array:");

		for(int i=0; i<size; i++ ) 
		{
			array[i] = sc.nextInt();
		}

		for (int i = 0 ;i< size-1; i++)
		{
			int min = i;

			for (int j = i+1; j<size; j++)
			{
				if (array[j] < array[min])
				{
					min = j;
				}
			}
			int temp = array[min];
			array[min] = array[i];
			array[i] = temp;
		}
		System.out.println("Sorted Array");
		for (int i = 0 ;i< size; i++)
		{
			System.out.print(" "+array[i]);
		}
	}  
}